lnwallet/chanfunding/wallet_assembler.go

```diff
@@ -285,16 +285,41 @@ func (w *WalletAssembler) ProvisionChannel(r *Request) (Intent, error) {
 		// If there's no funding amount at all (receiving an inbound
 		// single funder request), then we don't need to perform any
 		// coin selection at all.
-		case r.LocalAmt == 0:
+		case r.LocalAmt == 0 && r.FundUpToMaxAmt == 0 &&
+			r.FundMinAmt == 0:
 			break
 
+		// In case this request uses funding up to some maximum amount,
+		// we will call the specialized method for that.
+		case r.FundUpToMaxAmt != 0 && r.FundMinAmt != 0:
```

Comment:
should we catch the case where where one of these is set but the other is not? perhaps a sanity check somewhere? cause what could happen here is that a caller could set `LocalAmt` and _one of_ these fields which would then skip the previous case and this one and would end up in the default case

Reply:
Makes sense, I'll add an extra `case` for that.